tests/unit/test-firefox/test.preferences.js

```diff
@@ -34,4 +36,49 @@ describe('firefox/preferences', () => {
 
   });
 
+  describe('coerceCLICustomPreference', () => {
+
+    it('converts a single --pref cli option from string to object', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=true');
+      assert.isObject(prefs);
+      assert.equal(prefs['valid.preference'], true);
+    });
+
+    it('converts array of --pref cli option values into object', () => {
+      const prefs = coerceCLICustomPreference([
+        'valid.preference=true', 'valid.preference2=false',
+      ]);
+      assert.isObject(prefs);
+      assert.equal(prefs['valid.preference'], true);
+      assert.equal(prefs['valid.preference2'], false);
+    });
+
+    it('converts boolean values', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=true');
+      assert.equal(prefs['valid.preference'], true);
+    });
+
+    it('converts number values', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=455');
+      assert.equal(typeof prefs['valid.preference'], 'number');
+    });
+
+    it('converts float values', () => {
+      const prefs = coerceCLICustomPreference('valid.preference=4.55');
+      assert.equal(prefs['valid.preference'], '4.55');
+    });
+
+    it('does not allow certain default preferences to be customized', () => {
+      const prefs = coerceCLICustomPreference('xpinstall.signatures.required');
```

Comment:
how about exporting the array of the `non overridable` preferences from the preferences module, so that we can test that none of that array of preferences can be overridden?